sourcegraph/sourcegraph
# Search Results: Add new filters panel

## What have been done in this PR
This PR contains changes about filter panel UI. Most of these changes were implemented by design spec [here](https://www.figma.com/file/EPgl0qVbqGSumvf52brhda/Improving-search-results?type=design&node-id=431-55452&mode=design&t=JVOoFT0yVIjLTXz3-0). In a nutshell 
- New filter panel UI 
- Context-free filters (like first section with type filters)
- Context-based filters (like repositories, commits authors and symbol kinds)
- Improvements about caching system so changing filters won't trigger query twice

https://github.com/sourcegraph/sourcegraph/assets/18492575/42f4c101-702c-4340-ad21-ab24c3c878e3


## Follow-ups
- Static filters, at the moment filters like "Symbols kind", "Commit date" and "Authors" are generated on the client and therefore they don't have count info and it also possible that some of them may produce zero state. This is because currently we don't have them in our Search API. 
- Polishing designs details such icons and badges (in this PR I was focusing more on how filters work rather how they look, polishing will be done in a follow up PR to simplify tech and design review)
- Mobile layout for the search results page with new filters UI
- Removing feature flags (some flags about search results page will go away very soon)

## Feature flags setup

This feature eventually will be enabled by default but at the moment it's behind a feature flag `newSearchResultFiltersPanel`, also make sure that your other feature flag `newSearchResultsUI` is enabled too since new filters UI is added only to the most recent version of the search results page (new search results page feature flag will be removed in a follow up PR) 

## Test plan
- Check that you can see and interact with new filters panel if you have enabled `newSearchResultFiltersPanel: true` feature flag
- Check that the old panel got no regressions
